viewer/packages/pointclouds/src/PointCloudFactory.ts

```diff
@@ -38,30 +42,34 @@ export class PointCloudFactory {
     }
 
     if (geometry.cylinder) {
-      return new CylinderPrimitive(geometry.cylinder.centerA, geometry.cylinder.centerA, geometry.cylinder.radius);
+      const centerA = new THREE.Vector3().fromArray(geometry.cylinder.centerA);
+      const centerB = new THREE.Vector3().fromArray(geometry.cylinder.centerB);
+      return new CylinderPrimitive(centerA, centerB, geometry.cylinder.radius);
     }
 
     throw Error('Annotation geometry type not recognized');
   }
 
-  private async getAnnotations(modelIdentifier: CdfModelIdentifier): Promise<BoundingVolume[]> {
+  private async getAnnotations(modelIdentifier: CdfModelIdentifier): Promise<CdfPointCloudObjectAnnotation[]> {
     const modelAnnotations = await this._sdkPlayground!.annotations.list({
       filter: {
         // @ts-ignore
         annotatedResourceType: 'threedmodel',
         annotatedResourceIds: [{ id: modelIdentifier.modelId }]
-      }
-    });
+      },
+      limit: 1000
+    }).autoPagingToArray({ limit: Infinity });
```

Comment:
For the future, avoid `autoPaginToArray` and iterate over results as they arrive. I think there is a convenience function for that

Reply:
Okay - but what would be the benefit of doing that?

Reply:
`autoPagingToArray` will need to allocate an array to hold all elements. Also, it will not allow processing elements in batches as they arrive, postponing all processing until all elements are downloaded (which could take a while with O(100.000) elements).